File: src/main/java/milk/entitymanager/entity/Chicken.php
package milk.entitymanager.entity;

import cn.nukkit.item.Item;
import cn.nukkit.Player;
import cn.nukkit.event.entity.EntityDamageByEntityEvent;
import cn.nukkit.entity.Creature;

class Chicken extends Animal{
    const NETWORK_ID = 10;

    public width = 0.4;
    public height = 0.75;

    public function getName(){
        return "Chicken";
    }

    public function initEntity(){
        this.setMaxHealth(4);
        if(isset(this.namedtag.Health)){
            this.setHealth((int) this.namedtag["Health"]);
        }else{
            this.setHealth(this.getMaxHealth());
        }
        super.initEntity();
        this.created = true;
    }

    public function targetOption(Creature creature, distance){
    	if(creature instanceof Player)
        	return creature.isAlive() && !creature.closed && creature.getInventory().getItemInHand().getId() == Item.SEEDS && distance <= 49;
    	return false;
    }

    public function getDrops(){
        drops = [];
        if(this.lastDamageCause instanceof EntityDamageByEntityEvent){
            switch(Utils.rand(0, 2)){
                case 0 :
                    drops[] = Item.get(Item.RAW_CHICKEN, 0, 1);
                    break;
                case 1 :
                    drops[] = Item.get(Item.EGG, 0, 1);
                    break;
                case 2 :
                    drops[] = Item.get(Item.FEATHER, 0, 1);
                    break;
            }
        }
        return drops;
    }

}

File: src/main/java/milk/entitymanager/entity/animal/walking/Chicken.java
package milk.entitymanager.entity.animal.walking;

import cn.nukkit.item.Item;
import cn.nukkit.Player;
import cn.nukkit.event.entity.EntityDamageByEntityEvent;
import cn.nukkit.entity.EntityCreature;
import cn.nukkit.level.format.FullChunk;
import cn.nukkit.nbt.tag.CompoundTag;
import milk.entitymanager.util.Utils;

public class Chicken extends WalkingAnimal{
    public static final int NETWORK_ID = 10;

    public Chicken(FullChunk chunk, CompoundTag nbt){
        super(chunk, nbt);
    }

    @Override
    public int getNetworkId(){
        return NETWORK_ID;
    }

    @Override
    public float getWidth(){
        return 0.4f;
    }

    @Override
    public float getHeight(){
        return 0.75f;
    }

    public String getName(){
        return "Chicken";
    }

    public void initEntity(){
        super.initEntity();

        this.setMaxHealth(4);
    }

    public boolean targetOption(EntityCreature creature, double distance){
    	if(creature instanceof Player){
            Player player = (Player) creature;
            return player.isAlive() && !player.closed && player.getInventory().getItemInHand().getId() == Item.SEEDS && distance <= 49;
        }
    	return false;
    }

    public Item[] getDrops(){
        if(this.lastDamageCause instanceof EntityDamageByEntityEvent){
            switch(Utils.rand(0, 2)){
                case 0 :
                    return new Item[]{Item.get(Item.RAW_CHICKEN, 0, 1)};
                case 1 :
                    return new Item[]{Item.get(Item.EGG, 0, 1)};
                case 2 :
                    return new Item[]{Item.get(Item.FEATHER, 0, 1)};
            }
        }
        return new Item[0];
    }

}